src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/DynamicExpressionUtilities.cs

```diff
@@ -0,0 +1,289 @@
+                if (match.Groups["offset"].Success && match.Groups["option"].Success)
+                {
+                    var offsetString = match.Groups["offset"].Value;
+                    if (int.TryParse(offsetString, out int offset))
+                    {
+                        var optionString = match.Groups["option"].Value;
+                        if (currentDateTimeOffset.TryAddOffset(offset, optionString, out var newDateTimeOffset))
+                        {
+                            currentDateTimeOffset = newDateTimeOffset.Value;
+                        }
+                        else
+                        {
+                            return node.CreateBindingFailure(HttpDiagnostics.InvalidOption(expressionText, optionString));
+                        }
+                    }
+                    else
+                    {
+                        return node.CreateBindingFailure(HttpDiagnostics.InvalidOffset(expressionText, offsetString));
+                    }
+                }
+                string format;
+                var formatProvider = Thread.CurrentThread.CurrentUICulture;
+                var type = match.Groups["type"];
+
+                string text;
+                if (string.IsNullOrWhiteSpace(type.Value))
+                {
+                    text = currentDateTimeOffset.ToString();
+                }
+                else
+                {
+                    if (string.Equals(type.Value, "rfc1123", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // For RFC1123, we want to be sure to use the invariant culture,
+                        // since we are potentially overriding the format for local date time
+                        // we should explicitly set the format provider to invariant culture
+                        formatProvider = CultureInfo.InvariantCulture;
+                        format = "r";
+                    }
+                    else if (string.Equals(type.Value, "iso8601", StringComparison.OrdinalIgnoreCase))
+                    {
+                        format = "o";
+                    }
+                    else
+                    {
+                        // This substring exists to strip out the double quotes that are expected in a custom format
+                        format = type.Value.Substring(1, type.Value.Length - 2);
+                    }
+
+                    text = currentDateTimeOffset.ToString(format, formatProvider);
+                }
+
+                if (DateTimeOffset.TryParse(text, out _))
+                {
+                    return node.CreateBindingSuccess(text);
+                }
+            }
+            return node.CreateBindingFailure(HttpDiagnostics.IncorrectDateTimeFormat(expressionText, dateTimeType));
+        }
+
+        private static HttpBindingResult<object?> GetRandInt(HttpExpressionNode node, string text, Match match)
```

Comment:
```suggestion
        private static HttpBindingResult<object?> GetRandomInt(HttpExpressionNode node, string text, Match match)
```

It's a bit more readable when you use the full word.